README.md

```diff
@@ -7,25 +7,81 @@ This is the `hackage-server` code. This is what powers <http://hackage.haskell.o
 
 ## Installing dependencies
 
-`hackage-server` depends on `libgd` and `zlib`. You'll also need `libbrotli-dev` for enabling tests.
+`hackage-server` depends on `libgd`, `zlib`, and other system libraries. You'll also need `libbrotli-dev` for enabling tests.
 
-### [`nix develop`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-develop.html)
+You can use the Nix package manager to provide these dependencies, or install them manually.
 
-If you have the [Nix package manager](https://nixos.org/) installed, the easiest way to run `hackage-server` is by using `nix develop`. It should be unnecessary to install any dependencies manually. In this repository:
+### Using the [Nix package manager](https://nixos.org/) and provided [Nix Flake](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake.html)
+
+If you have the Nix package manager installed, you can build and run `hackage-server` without manually installing any dependencies.
+
+This uses `flake.nix`, implemented with [`srid/haskell-flake`](https://github.com/srid/haskell-flake).
+
+There are at least three ways to use this `flake.nix`. Clone this repository, enter the repository directory, then choose one of these options:
+
+#### [`nix develop`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-develop.html)
```

Comment:
I tried `nix develop` on my Mac Monterey and got this:
```
$ nix develop  --extra-experimental-features nix-command --extra-experimental-features flakes
do you want to allow configuration setting 'allow-import-from-derivation' to be set to 'true' (y/N)? y
do you want to permanently mark this value as trusted (y/N)? n
do you want to allow configuration setting 'extra-substituters' to be set to 'https://hackage-server.cachix.org/' (y/N)? y
do you want to permanently mark this value as trusted (y/N)? n
do you want to allow configuration setting 'extra-trusted-public-keys' to be set to 'hackage-server.cachix.org-1:iw0iRh6+gsFIrxROFaAt5gKNgIHejKjIfyRdbpPYevY=' (y/N)? y
do you want to permanently mark this value as trusted (y/N)? n
warning: ignoring untrusted substituter 'https://hackage-server.cachix.org/', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
error:
       … while calling the 'derivationStrict' builtin

         at //builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'ghc-shell-for-hackage-server-0.5.1-0'
         whose name attribute is located at /nix/store/v798823rqbsn3qx22rwayvcbiz5gn89x-source/pkgs/stdenv/generic/make-derivation.nix:300:7

       … while evaluating attribute '__impureHostDeps' of derivation 'ghc-shell-for-hackage-server-0.5.1-0'

         at /nix/store/v798823rqbsn3qx22rwayvcbiz5gn89x-source/pkgs/stdenv/generic/make-derivation.nix:433:7:

          432|       __propagatedSandboxProfile = lib.unique (computedPropagatedSandboxProfile ++ [ propagatedSandboxProfile ]);
          433|       __impureHostDeps = computedImpureHostDeps ++ computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps ++ __impureHostDeps ++ stdenv.__extraImpureHostDeps ++ [
             |       ^
          434|         "/dev/zero"

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: assertion '(stdenv).isLinux' failed

       at /nix/store/v798823rqbsn3qx22rwayvcbiz5gn89x-source/pkgs/os-specific/linux/kernel/generic.nix:71:1:

           70|
           71| assert stdenv.isLinux;
             | ^
           72|
```
So this does not seem to work on MacOS after all, or am I doing something wrong?  I am a total nix nitwit, so maybe I am missing something otherwise obvious.
I am on `nix 2.15.0`.

Reply:
Good catch, thanks.

I forgot to apply this fix https://github.com/haskell/hackage-server/pull/1219

Reply:
This comment specifically https://github.com/haskell/hackage-server/pull/1219#issuecomment-1597140858

Reply:
@andreasabel , please test 
```
nix develop  --extra-experimental-features nix-command --extra-experimental-features flakes
```
again on branch `master`.

The best solution is for the GitHub Action to check `nix develop`. This would check both Mac and Linux. However I don't know if that is possible.

Reply:
I merged the PR because apparently it was already broken for Mac, so merging it would at worst leave it broken.

Reply:
Okay, I think this reproduces the issue in a GitHub Action:

https://github.com/haskell/hackage-server/actions/runs/5946853679/job/16128037931?pr=1220

`nix flake check` checks `nix develop`, as @\srid explained to me here: https://github.com/srid/haskell-flake/discussions/172#discussioncomment-6219087

-----------

No, that warning message is not the cause.

If you want to remove that warning on your Mac, you need to edit the root-owned `nix.conf`. There are two `nix.conf`. One owned by root, and another by your user.

Here is the location of the former on Mac:  https://stackoverflow.com/a/63150105/1007926

You can add these settings:
```
trusted-users = [whoami username]

experimental-features = nix-command flakes
```

https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-trusted-users

Then restart the Nix daemon; unsure how to do on Mac. Or restart the whole system.